server/src/main/java/org/apache/iotdb/db/qp/physical/crud/QueryPlan.java

```diff
@@ -49,6 +49,7 @@
   private List<String> measurements; // for group by device sql, e.g. temperature
   private Map<String, Set<String>> measurementsGroupByDevice; // for group by device sql, e.g. root.ln.d1 -> temperature
   private Map<String, TSDataType> dataTypeConsistencyChecker; // for group by device sql, e.g. root.ln.d1.temperature -> Float
+  private Map<Path, TSDataType> dataTypeMapping = new HashMap<>(); // for group by device sql
```

Comment:
Is this ever used?

Reply:
I don't know. It's the new code I fetch from master.